tags/serializers.py

```diff
@@ -0,0 +1,70 @@
+from rest_framework import serializers
+from rest_framework.relations import PrimaryKeyRelatedField
+
+from tags.models import Tag, Namespace
+
+
+class NamespaceSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Namespace
+        fields = ("id", "name", "scope", "scoped_to")
+
+    def validate_scope(self, value):
+        if value not in Namespace.SCOPES.keys():
+            raise serializers.ValidationError("The scope isn't valid")
+        return value
```

Comment:
See my comment on the `Namespace` model, DRF will do it by itself if the model knows that `scope` must be a key of `Namespace.SCOPES`.

Reply:
I'm not sure I quite get it

Reply:
I mean, if you define the `scope` field of the `Namespace` model like this:
`scope = models.CharField(…, choices=SCOPES.keys())`
then DRF will enforce your `validate_scope` function by itself